packages/core/src/sync-store/postgres/store.ts

```diff
@@ -669,6 +677,225 @@ export class PostgresSyncStore implements SyncStore {
+          .execute();
+
+        if (rpcTransactions.length > 0) {
+          const transactions = rpcTransactions.map((transaction) => ({
+            ...rpcToPostgresTransaction(transaction),
+            chainId,
+          }));
+          await tx
+            .insertInto("transactions")
+            .values(transactions)
+            .onConflict((oc) => oc.column("hash").doNothing())
+            .execute();
+        }
+
+        if (rpcTransactionReceipts.length > 0) {
+          const transactionReceipts = rpcTransactionReceipts.map(
+            (rpcTransactionReceipt) => ({
+              ...rpcToPostgresTransactionReceipt(rpcTransactionReceipt),
+              chainId,
+            }),
+          );
+          await tx
+            .insertInto("transactionReceipts")
+            .values(transactionReceipts)
+            .onConflict((oc) => oc.column("transactionHash").doNothing())
+            .execute();
+        }
+
+        // Delete existing traces with the same `transactionHash`. Then, calculate "callTraces.checkpoint"
+        // based on the ordering of "callTraces.traceAddress" and add all traces to "callTraces" table.
+        const traceByTransactionHash: {
+          [transacionHash: Hex]: SyncCallTrace[];
+        } = {};
+        for (const trace of rpcTraces) {
+          if (traceByTransactionHash[trace.transactionHash] === undefined) {
+            traceByTransactionHash[trace.transactionHash] = [];
+          }
+          traceByTransactionHash[trace.transactionHash].push(trace);
+        }
+
+        for (const transactionHash of Object.keys(traceByTransactionHash)) {
+          const traces = await tx
+            .deleteFrom("callTraces")
+            .returningAll()
+            .where("transactionHash", "=", transactionHash as Hex)
+            .where("chainId", "=", chainId)
+            .execute();
+
+          (traces as Omit<InsertableCallTrace, "checkpoint">[]).push(
+            ...traceByTransactionHash[transactionHash as Hex].map((trace) => ({
+              ...rpcToPostgresTrace(trace),
+              chainId,
+            })),
+          );
+
+          // Use lexographical sort of stringified `traceAddress`.
+          traces.sort((a, b) => {
+            return a.traceAddress < b.traceAddress ? -1 : 1;
+          });
+
+          for (let i = 0; i < traces.length; i++) {
+            const trace = traces[i];
+            const checkpoint = encodeCheckpoint({
+              blockTimestamp: hexToNumber(rpcBlock.timestamp),
+              chainId: BigInt(chainId),
+              blockNumber: trace.blockNumber,
+              transactionIndex: BigInt(trace.transactionPosition),
+              eventType: EVENT_TYPES.callTraces,
+              eventIndex: BigInt(i),
+            });
+
+            trace.checkpoint = checkpoint;
+          }
+
+          await tx
+            .insertInto("callTraces")
+            .values(traces)
+            .onConflict((oc) => oc.column("id").doNothing())
```

Comment:
I don't think it's really possible to have a conflict here considering we just deleted above, but it's double checking.